src/shlex.ts

```diff
@@ -41,17 +42,29 @@ export function* split(str: string, opt?: ShlexOptions): Iterable<string> {
 
     if (quoteChar) {
       if (char === quoteChar) {
+        // Reached the end of a sub-quoted token.
+        if (subquote) {
+          subquote = false;
+          token = (token || '') + char;
+        }
         // Reached the end of a quoted token.
         quoteChar = undefined;
         continue;
       }
-
       // Another quoted char
       token = (token || '') + char;
       continue;
     }
 
     if (quoteChars.includes(char)) {
+      // Beginning of a sub-quoted token
+      if (i > 1 && str.substring(i - 1, i + 1) === "\\\"") {
```

Comment:
It seems like `"\\\'"` should be handled too, maybe just when when opt.mode == "posix".